validator-api/src/cache.rs

```diff
@@ -0,0 +1,34 @@
+use anyhow::Result;
+use mixnet_contract::MixNodeBond;
+use std::time::Instant;
+use validator_client::Client;
+
+pub struct MixNodeCache {
+    value: Vec<MixNodeBond>,
+    as_at: Instant,
+    validator_client: Client,
+}
+impl MixNodeCache {
+    pub fn init(value: Vec<MixNodeBond>, validator_client: Client) -> Self {
```

Comment:
Since presumably we will have identical cache for gateways (or maybe  they should be kept together?), should we make it own a `validator_client`?

Reply:
I've no strong opinions on this one. I had a more general Caching implementation but scrapped it as it was getting complicated. It is probably cleaner to have `MixNodeCache` make its own client tough